app/controllers/votes_controller.rb

```diff
@@ -0,0 +1,33 @@
+class VotesController < ApplicationController
+  before_action :find_micropost
+  
+  def create
+
+    if already_voted?
+      flash[:warning] = "You can't vote more than once"
+    else
+      # I know it is not safely because someone can hack database by coding in form (or somehow).
+      # Does validation in model (1..5) solve this problem ?
+     @vote = @micropost.votes.create(grades: params[:grades], user_id: current_user.id) 
+      if @vote.save
```

Comment:
A line higher you create a vote with a `create` method. 
`create` persists object to database, so it's equivalent to doing:
```
new_vote = @micropost.votes.build(grades: params[:grades], user_id: current_user.id) 
new_vote.save
```

So saving here has no use. You could use build+save method that i showed you or you could check with `@vote.errors.any?`